### ISSUE TITLE
Can't connect to SafePal Wallet on IOS

### ISSUE THREAD

**Issue by AmjadKhan2k18** [WalletConnect/WalletConnectFlutterV2#183]
I am trying to connect to SafePal wallet using universal link `https://link.safepal.io/` but it's not connecting, Connect Screen appear in SafePal wallet after clicking on approve button loading appear and nothing happen, on android it work

**To Reproduce**
try to connect to SafePal wallet in IOS

**Screenshots**
<img width="378" alt="Screenshot 2023-09-05 at 12 24 47 PM" src="https://github.com/WalletConnect/WalletConnectFlutterV2/assets/44894621/e0932bd2-7263-4b95-ae18-53308a66e643">

**Smartphone (please complete the following information):**
 - Device: Iphone 14, Iphone 11
 - OS: Latest

**Comment by quetool** [comment]
Hello @AmjadKhan2k18! I was able to reproduce this behavior also by scanning with SafePal wallet a QR code generated (outside WalletConnect) with the connection URI. Which makes me think it's probably an issue on SafePal side. I also tried an older version of iOS (15) and there I get an "Invalid Code" error when on SafePal side.
Also the fact that you are able to finish the connection with Android supports the theory that this is on SafePal side as we don't do anything really different between Android and iOS, but for sure we will keep investigating.
